concordium-sdk/src/main/java/com/concordium/sdk/responses/FinalizedBlockItemIterator.java

```diff
@@ -0,0 +1,140 @@
+package com.concordium.sdk.responses;
+
+import com.concordium.sdk.ClientV2;
+import com.concordium.sdk.responses.blocksatheight.BlocksAtHeightRequest;
+import com.concordium.sdk.types.AbsoluteBlockHeight;
+import com.concordium.sdk.types.UInt64;
+import lombok.val;
+import lombok.var;
+
+import java.util.Iterator;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An {@link Iterator} for {@link BlockIdentifier}s of finalized blocks. Contains a background task that
+ * polls for new finalized blocks indefinitely. This task can be stopped by calling {@link FinalizedBlockItemIterator#drop()}.
+ */
+public class FinalizedBlockItemIterator implements Iterator<BlockIdentifier> {
```

Comment:
Is the goal of this class to just poll blocks on a separate thread? I don't think we have an equivalent in the web-sdk, does the rust-sdk have something similar to this since this was implemented?

Reply:
The Rust SDK has https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/v2/struct.FinalizedBlocksStream.html which is returned by https://docs.rs/concordium-rust-sdk/latest/concordium_rust_sdk/v2/struct.Client.html#method.get_finalized_blocks_from

This is typically used by indexers to monitor the chain.

Reply:
Correct, I modeled this after the method linked above.